SR830/SR830.py

```diff
@@ -401,8 +426,124 @@ def output_overload(self, c):
         resp = yield dev.query("LIAS? 2")
         returnValue(bool(int(resp)))
 
+    @setting(99, 'Input Ground', gnd='i', returns='i')
+    def input_ground(self, c, gnd=None):
+        """Get or sets the input shield ground configuration.
+
+        Args:
+            gnd (int):  0: float; 1: ground
+        Returns:
+            (int):  The input shield ground status after setting (if requested).
+        """
+        dev = self.selectedDevice(c)
+        if gnd is not None:
+            gnd = int(gnd)
+            if gnd not in [0, 1]:
+                raise Exception('Error, requested input shield ground {}. '
+                                'Please select "0" for float or "1" for ground.'
+                                ''.format(gnd))
+            yield dev.write("IGND {}".format(gnd))
+        resp = yield dev.query("IGND?")
+        returnValue(int(resp))
+
+
+    @setting(37, 'Input Coupling', coupling='s', returns='i')
+    def input_coupling(self, c, coupling=None):
+        """Get or sets the input coupling.
+
+        Args:
+            coupling:  0: AC or 1: DC
+        Returns:
+            (int):  The input coupling after setting (if requested).
+        """
+        coupling_dict = {
+            'AC':0,
+            'DC':1
+        }
+        dev = self.selectedDevice(c)
+        if coupling is not None:
+            if isinstance(coupling, str):
+                coupling = coupling.upper()
+                if (coupling not in coupling_dict.keys()) and (coupling not in [0,1]):
```

Comment:
can replace this line with:

If coupling not in (coupling_dict.keys() + [0,1]) to make it shorter